### DISCUSSION TITLE
Force the use of Qt backend on macOS with C++

### DISCUSSION THREAD

**Discussion by aydwi** [slint-ui/slint#1389] [Q&A]
Hi! I'd like to force the use of Qt backend on macOS while trying out the sample apps with C++ and CMake. I have both Qt and OpenGL available in my system library, but my compiled apps always seem to come on looking like Windows-flavour. I tried `export DYLD_FRAMEWORK_PATH="/usr/local/opt/qt@5/lib"` but that did not help. Any ideas?

<img width="612" alt="image" src="https://user-images.githubusercontent.com/29029116/177249026-0ab8554a-0418-4668-b9ff-463a4b3bc9a2.png">

**Comment by ogoffart** [comment]
I'm not an expert in macOS, but i believe this should work if cmake finds Qt.  So you may need to set some cmake variable like CMAKE_PREFIX_PATH or so, while running cmake

**Reply by tronical** [reply]
You could try to configure your CMake build with `-DSLINT_FEATURE_BACKEND_GL_ALL=OFF` `-DSLINT_FEATURE_BACKEND_QT=ON` to force disabling of the GL backend and enabling of Qt.

**Reply by aydwi** [reply]
`-DSLINT_FEATURE_BACKEND_GL_ALL=OFF` `-DSLINT_FEATURE_BACKEND_QT=ON` definitely did the trick, however I ended up with a situation where the executable panicked at runtime due to "no backend being available", presumably because CMake was not able to find qt5 even though I set `CMAKE_PREFIX_PATH` and several other envirnonment variables.

The fix for me was to install **qt6**, after which I was able to obtain the native Cocoa/Appkit based UI. I will close this discussion now, but perhaps it is a good idea to include this information in the README. I think a lot of macOS developers will be looking for a native UI when testing/using Slint.

As a side note, sincere thanks for your work @tronical and @ogoffart. I read some of the C++ code (the parts I could understand anyway), and quickly realized how herculean of an effort Slint is. Interestingly, I worked on a similar project in the past two years (very similar in code generation technique, although not in the GUI space). Thank you for your hard work over the years and offering this to the community for free.
